src/test/java/com/amazon/opendistroforelasticsearch/sql/esdomain/mapping/FieldMappingTest.java

```diff
@@ -0,0 +1,101 @@
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.esdomain.mapping;
+
+import com.amazon.opendistroforelasticsearch.sql.domain.Field;
+import com.amazon.opendistroforelasticsearch.sql.util.MatcherUtils;
+import org.elasticsearch.action.admin.indices.mapping.get.GetFieldMappingsResponse.FieldMappingMetaData;
+import org.hamcrest.Matcher;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyMap;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Unit test for {@code FieldMapping} with trivial methods ignored such as isSpecified, isMetaField etc.
+ */
+public class FieldMappingTest {
+
+    @Test
+    public void testFieldMatchesWildcardPatternSpecifiedInQuery() {
+        assertThat(
+            new FieldMapping("employee.first", typeMappings(), fieldsSpecifiedInQuery("employee.*")),
+            isWildcardSpecified(true)
+        );
+    }
+
+    @Test
+    public void testFieldMismatchesWildcardPatternSpecifiedInQuery() {
+        assertThat(
+            new FieldMapping("employee.first", typeMappings(), fieldsSpecifiedInQuery("manager.*")),
+            isWildcardSpecified(false)
+        );
+    }
+
+    @Test
+    public void testFieldIsProperty() {
+        assertThat(
+            new FieldMapping("employee.first"),
+            isPropertyField(true)
+        );
+
+        assertThat(
+            new FieldMapping("employee.first.keyword"),
+            isPropertyField(true)
+        );
+    }
+
+    @Test
+    public void testFieldIsNotProperty() {
+        assertThat(
+            new FieldMapping("employee"),
+            isPropertyField(false)
+        );
+
+        assertThat(
+            new FieldMapping("employee.keyword"),
+            isPropertyField(false)
+        );
+    }
+
+    private Matcher<FieldMapping> isWildcardSpecified(boolean isMatched) {
+        return MatcherUtils.featureValueOf("is field match wildcard specified in query",
+                                           is(isMatched),
+                                           FieldMapping::isWildcardSpecified);
+    }
+
+    private Matcher<FieldMapping> isPropertyField(boolean isProperty) {
+        return MatcherUtils.featureValueOf("isPropertyField",
+                                           is(isProperty),
+                                           FieldMapping::isPropertyField);
+    }
+
+    private Map<String, Field> fieldsSpecifiedInQuery(String...fieldNames) {
+        return Arrays.stream(fieldNames).
+                      collect(Collectors.toMap(name -> name,
+                                               name -> new Field(name, "")));
+    }
+
+    private Map<String, FieldMappingMetaData> typeMappings() {
```

Comment:
Please inline this method, it just makes test less understandable

Reply:
done! Was planning to use this for more test cases against `type()` and then gave up due to time constraint...